static-build/cmake/AddDependencyProjects.cmake

```diff
@@ -33,14 +33,20 @@ set(BACKUP_STORAGE https://distrib.hb.bizmrg.com)
 # using of corresponding environment variable. It is possible that a
 # linker flag assumes that some compilation flag is set. We don't pass
 # CFLAGS from environment, so we should not do it for LDFLAGS too.
-set(DEPENDENCY_CFLAGS "")
-set(DEPENDENCY_CXXFLAGS "")
+#
+# Note: A configure script generated by autoconf doesn't enable compiler
+# optimizations if CFLAGS / CXXFLAGS is set so we have to enable them
+# explicitly, see https://github.com/tarantool/tarantool/issues/8606
+#
+set(DEPENDENCY_CFLAGS "-O2")
+set(DEPENDENCY_CXXFLAGS "-O2")
```

Comment:
Should we also add "-g" for debug build?

Reply:
There's no such option in Tarantool static-build CMake config as "debug build". Also, we never added `-g` to dependency projects. I don't think that we need to have debug info for them because we don't expect bugs in dependencies.

Reply:
> we don't expect bugs in dependencies

On the one hand you are right but on the other hand it's still possible and in case if it happens one day it could be quite painful.

Well, let's stay as is.